3574/1/top_gems.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'parse'
+require_relative 'option_parser'
+require_relative 'read_file'
+require_relative 'table'
+require_relative 'option_functions'
+
+HEADER = %i[name_id used_by watch stars fork issues contributors].freeze
+
+class TopGems
+  def initialize
+    @gems_from_file = {}
+    @gems_from_file[:file_name] = 'list_gems.yml'
+    @gems_from_file[:number] = []
+    @gem_data = {}
+  end
+
+  def call
+    Options.new.call(@gems_from_file)
+    names = ReadFile.new.call(@gems_from_file)
+    @gem_data = InfoParser.new.take_information(names)
+    @gem_data = TopFunction.new.call(@gem_data, @gems_from_file)
```

Comment:
Your method from TopFunction class returned reorder list of gems and probably it is shorter than it was. You can't name it the same way as it was right after reading from file